includes/class-freemius.php

```diff
@@ -629,6 +636,254 @@ private function _version_updates_handler() {
             }
         }
 
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since  2.1.0
+         *
+         * @param string $sdk_prev_version
+         * @param string $sdk_version
+         */
+        function _handle_gdpr_admin_notice( $sdk_prev_version, $sdk_version ) {
+            /**
+             * @since 1.1.7.3 Fixed unwanted connectivity test cleanup.
+             */
+            if ( empty( $sdk_prev_version ) ) {
+                return;
+            }
+
+            if ( version_compare( $sdk_prev_version, '2.1.0', '<' ) &&
+                version_compare( $sdk_version, '2.1.0', '>=' )
+            ) {
+                add_action( 'init', array( &$this, '_maybe_show_gdpr_admin_notice' ) );
+            }
+        }
+
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since  2.1.0
+         */
+        function _maybe_show_gdpr_admin_notice() {
+            if ( ! $this->is_user_in_admin() ) {
+                return;
+            }
+
+            self::require_pluggable_essentials();
+
+            if ( ! $this->is_user_admin() ) {
+                return;
+            }
+
+            $storage         = FS_Storage::instance( 'gdpr_global', '' );
+            $current_wp_user = self::_get_current_wp_user();
+
+            self::$_all_admin_notices = FS_Admin_Notices::instance( 'all_admins', '', '', true );
+
+            if ( self::$_all_admin_notices->has_sticky( "gdpr_optin_actions_{$current_wp_user->ID}", true ) ) {
+                return;
+            }
+
+            $show_gdpr_optin_notice = $storage->get( "show_gdpr_optin_notice_{$current_wp_user->ID}", true );

```

Comment:
@fajardoleo this is a gdpr focused storage, so there's no need to specify that in the var names. You can use `show_notice_{$current_wp_user->ID}` instead.